guardrails/validator_base.py

```diff
@@ -167,8 +171,36 @@ class Validator:
     _metadata = {}
 
     def __init__(
-        self, on_fail: Optional[Union[Callable, OnFailAction]] = None, **kwargs
+        self,
+        use_local: bool = None,
+        validation_endpoint: str = None,
```

Comment:
These should be optional, and since they're optional it's safer to declare them as keyword-only arguments.  This way the positional arguments in the method signature can stay the same.